### PR TITLE
Remove "paranoid" setting from db models (exc. Users & Profiles)

### PR DESCRIPTION
**Pull request by iamigo**
Remove isDeleted attribute (exc. Users & Profiles)
Add migration to clean up the soft-deleted data and update the indices
Update tests

Note: we are keeping paranoid on for Users because we can't delete some of the soft-deleted user records due to referential integrity, where another record has a soft-deleted user associated with it as creator and/or owner. And we're keeping it on for Profiles because we're keeping it on for Users.

Changes to /api/v1/helpers/nouns/*:

pull "isDeleted" out of the list of "readOnlyFields"
Changes to /db/model/*:

remove "paranoid" setting on model
remove "isDeleted" field defn
no longer call setIsDeleted in beforeDestroy
remove "isDeleted" from list of fields in any index defns
remove "isDeleted" from list of "ignoreAttributes" when calling publishObject

Migration:

Up
- fixReferences - clear the collectorGroupId col for collectors which reference a soft-deleted collector group
- destroySoftDeleted - deletes any soft-deleted records from the db
- removeOldIndices - deletes the indices which include deletedAt/isDeleted
- removeFields - removes the deletedAt/isDeleted columns

Down
- recreateOldFields - adds back the deletedAt/isDeleted columns
- removeNewIndices - removes the new indices without deletedAt/isDeleted, if they were created

### CODE REVIEW

**Comment by iamigo** [comment]
FYI I have tested in devbox with copy of prod db

File: tests/db/model/sample/timeout.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ describe('tests/db/model/sample/timeout.js >', () => {
     .catch(done);
   });
 
-  it('simulate 10 seconds in the past', (done) => {
+  it('simulate 1tests/db/model/subject/delete.js0 seconds in the past', (done) => {
```
Comment (pallavi2209) [inline tests/db/model/sample/timeout.js]: mistyped?

**Comment by pallavi2209** [comment]
@iamigo Just thinking.. do we need any changes for the data existing in Redis?

**Comment by iamigo** [comment]
@pallavi2209 no, I don't think so, since we delete records from redis when the postgres record is soft-deleted

File: migrations/20190603225918-no-paranoid.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+  /*
+   * Why this order?
+   * - Generator THEN Collector THEN CollectorGroup so we don't try to delete
+   *   a CollectorGroup which is referenced by a Collector or Generator
+   * - Generator THEN GeneratorTemplate so we don't try to delete a
+   *   GeneratorTemplate which is referenced by a Generator
+   * - Perpsective THEN Lens/Subject so we don't try to delete a Lens or
+   *   Subject which is referenced by a Perspective
+   */
```
Comment (jgraff2) [inline migrations/20190603225918-no-paranoid.js]: >> Promise.all will run the promises in parallel, so it's possible they won't be executed in the right order.

> Yeah at one point I had a sequence of Promise.alls but then I wrapped the whole list in a trx so it all happens atomically.

Does that mean the order doesn't matter here anymore? If so can you update the comment?

**Pull request merged**